chemprop/models/model.py

```diff
@@ -1,160 +1,341 @@
-from typing import List, Union, Tuple
+from argparse import Namespace
+from typing import List, Optional, Union, Tuple
+import warnings
 
 import numpy as np
 from rdkit import Chem
 import torch
 import torch.nn as nn
 
-from .mpn import MPN
 from chemprop.args import TrainArgs
 from chemprop.features import BatchMolGraph
 from chemprop.nn_utils import get_activation_function, initialize_weights
+from chemprop.models.mpn import MPN
 
+class nn_exp(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
 
-class MoleculeModel(nn.Module):
+    def forward(self, x):
+        return torch.exp(x)
+
+class _MoleculeModel(nn.Module):
     """A :class:`MoleculeModel` is a model which contains a message passing network following by feed-forward layers."""
 
-    def __init__(self, args: TrainArgs):
-        """
-        :param args: A :class:`~chemprop.args.TrainArgs` object containing model arguments.
-        """
-        super(MoleculeModel, self).__init__()
+    def __init__(
+        self,
+        dataset_type: str,
+        num_tasks: int,
+        multiclass_num_classes: int,
+        atom_messages: bool = False,
+        hidden_size: int = 300,
+        bias: bool = False,
+        depth: int = 3,
+        dropout: float = 0.0,
+        undirected: bool = False,
+        device: Union[str, torch.device] = "cpu",
+        aggregation: str = "mean",
+        aggregation_norm: int = 100,
+        activation: str = "ReLU",
+        atom_descriptors: Optional[str] = None,
+        atom_descriptors_size: int = 0,
+        features_only: bool = False,
+        use_input_features: bool = False,
+        overwrite_default_atom_features: bool = False,
+        overwrite_default_bond_features: bool = False,
+        mpn_shared: bool = False,
+        number_of_molecules: int = 1,
+        checkpoint_frzn: Optional[str] = None,
+        freeze_first_only: bool = False,
+        frzn_ffn_layers: int = 0,
+        features_size: Optional[int] = None,
+        ffn_num_layers: int = 2,
+        ffn_hidden_size: Optional[int] = None,
+        spectra_activation: str = "exp",
+        **kwargs,
```

Comment:
Is there a way to set this up to enforce that the defaults in the args file match the defaults here? We need defaults here so that people can reasonably use this function outside of the whole workflow. I worry a little about having outdated defaults that differ between the two locations.

Reply:
yeah I tried my best to match them 1:1 but I can go back and double check

Reply:
I'm sure you matched them right. I was thinking about future changes. Not sure if there's a reasonable tool that would do that, probably not.